src/uint/mul.rs

```diff
@@ -86,7 +86,73 @@ impl<const LIMBS: usize> UInt<LIMBS> {
 
     /// Square self, returning a "wide" result in two parts as (lo, hi).
     pub const fn square_wide(&self) -> (Self, Self) {
-        self.mul_wide(self)
+        // This is a re-implementation from https://github.com/ucbrise/jedi-pairing/blob/c4bf151b8d2b560973cb3805f9b5f4a144597f7e/include/core/bigint.hpp#L410.
```

Comment:
We try to be pretty careful with IPR, even with source code translations, with the goal of being able to confidently state the resulting code is Apache 2+MIT.

This means we generally prefer public domain sources or getting an explicit blessing from the original author to license the resulting work as Apache 2+MIT.

Reply:
@samkumar are you OK with this translation of your code being licensed as Apache 2.0+MIT? If so, would you prefer a larger IPR statement about the original code?

Reply:
I'm OK with it being licensed as Apache 2.0 + MIT. Can you clarify what you mean by "a larger IPR statement about the original code"?

Reply:
@samkumar awesome, thanks!

The statement is whatever you'd like us to put, just let us know. Is the existing comment OK?

We can probably also mention permission was given on this PR.

Reply:
Updated the comment about the original implementation with relicensing info in this commit: [a158037](https://github.com/RustCrypto/crypto-bigint/pull/133/commits/a1580376553d54f9e73a3f6de7fec7cec66a02bc)